sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/CustomTimestampPolicyWithLimitedDelay.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import java.util.Optional;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.windowing.BoundedWindow;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.annotations.VisibleForTesting;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+/**
+ * A policy for custom record timestamps where timestamps within the queue are expected to be
+ * roughly monotonically increasing with a cap on out of order event delays (say 1 minute). The
+ * watermark at any time is '({@code Min(now(), Max(event timestamp so far)) - max delay})'.
```

Comment:
```suggestion
 * watermark at any time is '({@code earliest(now(), latest(event timestamp so far)) - max delay})'.
```

I like to use "earliest" and "latest" rather than "min" and "max" when comparing time

(whereas "max delay" is a param so that stays)

Reply:
(there's javadoc below with a similar pseudocode)